src/bundle/Controller/LocationController.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/**
+ * File containing the LocationController class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\HybridPlatformUiBundle\Controller;
+
+use eZ\Publish\Core\MVC\Symfony\Controller\Controller;
+use eZ\Publish\API\Repository\Values\Content\Content;
+use EzSystems\HybridPlatformUi\Form\UiFormFactory;
+use EzSystems\HybridPlatformUi\Repository\UiLocationService;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\RouterInterface;
+
+class LocationController extends Controller
+{
+    public function actionsAction(
+        Content $content,
+        Request $request,
+        UiLocationService $uiLocationService,
+        RouterInterface $router,
+        UiFormFactory $formFactory
+    ) {
+        $actionsForm = $formFactory->createLocationsActionForm();
+        $actionsForm->handleRequest($request);
+
+        if ($actionsForm->isValid()) {
+            $locationIds = array_keys($actionsForm->get('removeLocations')->getData());
+
+            if ($actionsForm->get('delete')->isClicked()) {
+                $uiLocationService->deleteLocations($locationIds, $content->id);
+            }
+        }
+
+        return new RedirectResponse(
+            $router->generate(
+                '_ez_content_view',
+                [
+                    'contentId' => $content->id,
```

Comment:
One difficulty with that is that it changes the logic behind `UiLocation::canRemove()`. Each location needs to know about the currently viewed location, passed as a controller argument.

Checking if a location is the currently viewed one requires that we have this value for comparison, either in the controller, or the template, or in the service. In addition, it may be needed when removing locations, if the `UiLocationService::deleteLocations()` method is meant to enforce that logic.

One suggestion, after digging through the code: change `loadLocations()` to load the locations _given a Location_ (the currently viewed one, that you get from `$view`. That way, we know which one the loading is done from, and we can use that to customize the location object. Then I'd also change `deleteSecondaryLocations()` to take the location we delete from as an argument, so that we can do the same check. This service is the one that manipulates locations from a UI's perspective, it is normal that it has the context it needs to implement the business logic.

The `UiService` turns out to be more of a `LocationTabService`, given how specialized it is, or `UiLocationList`.